buildingspy/development/check_parameters.py

```diff
@@ -0,0 +1,476 @@
+# If any of the checks fail, the script returns with an error.
+#
+# TSNouidui@lbl.gov                            2017-01-24
+###########################################################
+#
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import logging as log
+
+log.basicConfig(filename='check_parameters.log', filemode='w',
+                level=log.DEBUG, format='%(asctime)s %(message)s',
+                datefmt='%m/%d/%Y %I:%M:%S %p')
+stderrLogger = log.StreamHandler()
+stderrLogger.setFormatter(log.Formatter(log.BASIC_FORMAT))
+log.getLogger().addHandler(stderrLogger)
+
+import os
+import re
+import sys
+
+class Validator(object):
+    ''' Class that validates ``.mo`` files for the correct syntax.
+    '''
+    def __init__(self):
+
+        # --------------------------
+        # Class variables
+        self._rewriteModel = False
+
+    def _recursive_glob(self, rootdir='.', suffix=''):
+        return [os.path.join(rootdir, filename) for rootdir, dirnames,
+                filenames in os.walk(rootdir) for filename in filenames 
+                if (filename.endswith(suffix) 
+                    and ("ConvertBuildings_from" not in filename)) ]    
+    
+    def _check_experiment(self, name, val, value, modelPath, mos_file):
+        """ 
+        Check experiment annotation in mo file.
+    
+        :param name: Parameter name.
+        :param val: Value found in mo file.
+        :param value: Value found in mos file.
+        :param modelPath: Path to mo file.
+        :param mos_file: Path to mos file.
+    
+         """
+        
+        if("*" in str(val)):
+            log.error("Found mo_file: {!s} with **experiment** annotation: {!s} which contains the invalid expression: {!s}."
+                      .format(modelPath, self._capitalize_first(name), str(val)))
+            sys.exit(1)
+    
+        delta = abs(eval(val) - eval(value))
+        
+        if (delta!=0):
+            log.error("Found mo_file: {!s} with **experiment** annotation: **{!s}**.".format(modelPath, self._capitalize_first(name)))
+            log.error("The value of **{!s}** is : {!s} which is different from the (default) value: {!s} found in the .mos file: {!s}."
+                      .format(val, value, name, mos_file))
+            sys.exit(1)
+            
+    def _missing_parameter(self, name, value, modelPath, mos_file):
+        """ 
+        Check experiment annotation in mo file.
+    
+        :param name: Parameter name.
+        :param value: Value found in mos file.
+        :param modelPath: Path to mo file.
+        :param mos_file: Path to mos file.
+    
+         """
+        
+        log.error("Found mo_file: {!s} without **experiment** annotation **{!s}**.".format(modelPath, self._capitalize_first(name)))
+        log.error("The parameter **"+name+"** with value : {!s} is however defined in the .mos file {!s}.".format(value, mos_file)) 
+        sys.exit(1)   
```

Comment:
Raise an exception, but never call `sys.exit(1)` as raising exceptions allows the error to be handled.